code/game/objects/structures/salvaging.dm

```diff
@@ -0,0 +1,649 @@
+/obj/structure/salvageable
+	name = "broken machinery"
+	desc = "It's broken beyond repair. You may be able to salvage something from this."
+	icon = 'icons/obj/salvage_structure.dmi'
+	density = TRUE
+	anchored = TRUE
+	var/salvageable_parts = list()
+
+/obj/item/stack/ore/salvage/examine(mob/user)
+	. = ..()
+	. += "You can use a crowbar to salvage this."
+
+/obj/structure/salvageable/proc/dismantle(mob/living/user)
+	var/obj/frame = new /obj/structure/frame/machine(get_turf(src))
+	frame.anchored = TRUE
+	for(var/path in salvageable_parts)
+		if(prob(salvageable_parts[path]))
+			new path (loc)
+	return
+
+/obj/structure/salvageable/attackby(obj/item/tool, mob/living/user, params)
+	if((user.a_intent != INTENT_HELP) && tool.tool_behaviour == TOOL_CROWBAR)
+		return
+	else
+		return ..()
```

Comment:
you dont need an else here, you already returned if failing

Reply:
moved the return